plugins/it.xsemantics.dsl/src/it/xsemantics/dsl/XsemanticsModelGenerator.xtend

```diff
@@ -0,0 +1,47 @@
+package it.xsemantics.dsl
+
+import java.util.List
+import org.eclipse.xtend2.lib.StringConcatenation
+import org.eclipse.xtext.common.types.JvmDeclaredType
+import org.eclipse.xtext.generator.trace.LocationData
+import org.eclipse.xtext.nodemodel.INode
+import org.eclipse.xtext.util.ITextRegionWithLineInformation
+import org.eclipse.xtext.xbase.compiler.FileHeaderAdapter
+import org.eclipse.xtext.xbase.compiler.GeneratorConfig
+import org.eclipse.xtext.xbase.compiler.JvmModelGenerator
+import org.eclipse.xtext.xbase.compiler.output.ITreeAppendable
+import org.eclipse.xtext.xbase.lib.util.ReflectExtensions
+import com.google.inject.Inject
+import org.eclipse.xtext.documentation.IFileHeaderProvider
+
+class XsemanticsModelGenerator extends JvmModelGenerator {
+	@Inject protected extension ReflectExtensions
+
+	override void generateFileHeader(JvmDeclaredType declType, ITreeAppendable appendable, GeneratorConfig config) {
+		val fileHeaderAdapter = declType.eAdapters.filter(FileHeaderAdapter).head
+		if (!fileHeaderAdapter?.headerText.nullOrEmpty) {
+			val fileHeaderProvider = get('fileHeaderProvider') as IFileHeaderProvider
+			val fileHeaderNodes = fileHeaderProvider.getFileHeaderNodes(declType.eResource)
```

Comment:
True. But the superclass method implementation is analog to this one here, so I don't see a need to treat it differently here.